polylogue/cli/imports.py

```
@@ -241,8 +270,18 @@ def run_import_chatgpt(args: argparse.Namespace, env: CommandEnv) -> None:
             entries = list_chatgpt_conversations(export_path)
         except Exception as exc:
-            console.print(f"[red]Failed to scan export: {exc}")
-            return
+            json_mode = getattr(args, "json", False)
+            if json_mode:
+                raise JSONModeError(
+                    "invalid_export",
+                    f"Failed to scan ChatGPT export: {exc}",
+                    path=str(export_path),
+                    hint="Ensure the export file is a valid ChatGPT conversations.json or .zip export",
+                )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)